docs/discussions.rst

```diff
@@ -149,7 +146,39 @@ and you don't want to have to maintain their equivalent using a synchronous inte
 Procrastinate supports asynchronous job deferring, and asynchronous job execution,
 either serial or parallel (see `howto/async`, `howto/concurrency`).
 
-The rest of this section will discuss the specifics of asynchronous parallel workers.
+.. _discussion-sync-defer:
+
+Synchronous deferring
+^^^^^^^^^^^^^^^^^^^^^
+
+Procrastinate gets to be called in two very different contexts:
+
+- When deferring a task, in **your** process, where Procrastinate is being used as a
+  library
+- When running a worker, where Procrastinate itself controls the process.
+
+Workers will always be asynchronous, and will support both synchronous and asynchronous
+jobs.
+
+When deferring a task, on the other hand, Procrastinate needs to play nice with
+your program. This is where there are choices to make.
+
+Procrastinate supports two ways of doing synchronous I/O:
+
+- "classic" synchronous I/O (using synchronous database drivers such as ``Psycopg2``).
+  This mode is necessary in multi-threaded cases.
+- "mixed" I/O (synchronously launching an event loop, and have asynchronous coroutine
+  run under the hood).
+  This mode will use less connections when used within another job.
```

Comment:
I've been thinking about this whole thing over the week-end :smile: And I must say that I am not a big fan of supporting two ways of doing synchronous I/O. The paragraph above, where we explain that there's a "classic" way and a "mixed" way, is, to me, a clear sign of complexity and poor design. I think what I'd like to see a very clear separation between the sync API and the async API. I'd even consider removing the `add_sync_api` decorator, and making the async API async only. And introducing a separate sync API for sync applications. I know my words may sound a bit theoretic… but that's all I have at this point.

Reply:
Ok, but what about deferring from within a job ? You already have an app and you most probably wish not to make a new pool for this.